agent/api/task_routes.py

```
@@ -172,8 +172,42 @@ async def get_recent_failed_tasks(
 
+    @router.get("/task-suppressions", response_model=List[TaskSuppressionRule])
+    async def list_task_suppressions(
+        include_expired: bool = False,
+        session: Session = Depends(get_db),
+    ):
+        return SuppressionService(session).list_rules(include_expired=include_expired)
+
+    @router.post("/task-suppressions", response_model=TaskSuppressionRule)
+    async def create_task_suppression(
+        payload: TaskSuppressionRuleCreate,
+        session: Session = Depends(get_db),
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)